WcaOnRails/spec/controllers/api_competitions_controller_spec.rb

```diff
@@ -40,4 +40,234 @@
       expect(parsed_body["website"]).to eq "http://example.com"
     end
   end
+
+  describe 'GET #index' do
+    it 'sorts newest to oldest' do
+      yesteryear_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.year.ago)
+      yesterday_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.ago)
+      today_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 0.days.ago)
+      tomorrow_comp = FactoryGirl.create(:competition, :confirmed, :visible, starts: 1.day.from_now)
+
+      get :index
+      expect(response.status).to eq 200
+      json = JSON.parse(response.body)
+      expect(json.map { |c| c["id"] }).to eq [tomorrow_comp, today_comp, yesterday_comp, yesteryear_comp].map(&:id)
```

Comment:
I didn't realize `v0` was meant to signal a preliminary in flux state. If that's the intention, the current testing is fine. To me, "version 1.0 alpha" sounds like something that aspires to be a real API soon.

That ambition level could possibly be made more obvious somehow. No I don't know how :)